src/main/java/io/cdap/plugin/dlp/SensitiveRecordFilter.java

```diff
@@ -0,0 +1,319 @@
+import com.google.cloud.dlp.v2.DlpServiceClient;
+import com.google.cloud.dlp.v2.DlpServiceSettings;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.privacy.dlp.v2.ContentItem;
+import com.google.privacy.dlp.v2.GetInspectTemplateRequest;
+import com.google.privacy.dlp.v2.InspectContentRequest;
+import com.google.privacy.dlp.v2.InspectContentResponse;
+import com.google.privacy.dlp.v2.InspectResult;
+import com.google.privacy.dlp.v2.InspectTemplate;
+import com.google.privacy.dlp.v2.ProjectName;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.InvalidEntry;
+import io.cdap.cdap.etl.api.MultiOutputEmitter;
+import io.cdap.cdap.etl.api.MultiOutputPipelineConfigurer;
+import io.cdap.cdap.etl.api.MultiOutputStageConfigurer;
+import io.cdap.cdap.etl.api.SplitterTransform;
+import io.cdap.cdap.etl.api.StageMetrics;
+import io.cdap.cdap.etl.api.StageSubmitterContext;
+import io.cdap.cdap.etl.api.TransformContext;
+import io.cdap.cdap.format.StructuredRecordStringConverter;
+import io.cdap.plugin.gcp.common.GCPConfig;
+import io.cdap.plugin.gcp.common.GCPUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * This class <code>SensitiveRecordFilter</code> provides an easy way to filter sensitive PII data from stream. The
+ * class utilizes Data Loss Prevention APIs for identifying sensitive data. Depending on the filter confidence set by
+ * user, the class either sends input record on sensitive port or non-sensitive port.
+ *
+ * <p>
+ * In case of issue with invoking DLP, the plugin depending on user choice either chooses to skip record, error pipeline
+ * or send record to error port.
+ * </p>
+ */
+@Plugin(type = SplitterTransform.PLUGIN_TYPE)
+@Name(SensitiveRecordFilter.NAME)
+@Description(SensitiveRecordFilter.DESCRIPTION)
+public final class SensitiveRecordFilter extends SplitterTransform<StructuredRecord, StructuredRecord> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SensitiveRecordFilter.class);
+  public static final String NAME = "SensitiveRecordFilter";
+  public static final String DESCRIPTION = "Filters input records based that are sensitive.";
+  private static final String SENSITIVE_PORT = "Sensitive";
+  private static final String NON_SENSITIVE_PORT = "Non-Sensitive";
+
+  // Stores the configuration passed to this class from user.
+  private final Config config;
+
+  private StageMetrics metrics;
+
+  // DLP service client for managing interactions with DLP service.
+  private DlpServiceClient client;
+
+  @VisibleForTesting
```

Comment:
since there are no tests going in with this PR, we can remove this

Reply:
done